src/index.js

```diff
@@ -202,172 +286,266 @@ class PeerId {
-    throw new Error('Public and private key do not match')
-  }
+  /**
+   * Create PeerId from Protobuf bytes. 
+   * 
+   * @param {Uint8Array | string} buf - Protobuf bytes, as Uint8Array or hex-encoded string.
+   * @returns {Promise<PeerId>}
+   */
+  static async createFromProtobuf (buf) {
+    if (typeof buf === 'string') {
+      buf = uint8ArrayFromString(buf, 'base16')
+    }
 
-  if (id && !uint8ArrayEquals(privDigest, id)) {
-    throw new Error('Id and private key do not match')
-  }
+    let { id, privKey: privKeyArray, pubKey: pubKeyArray } = PeerIdProto.decode(buf)
 
-  return new PeerIdWithIs(id, privKey, pub)
-}
+    /**
+     * @type {PrivateKey | false}
+     */
+    const privKey = privKeyArray ? await cryptoKeys.unmarshalPrivateKey(privKeyArray) : false
 
-exports.createFromProtobuf = async (buf) => {
-  if (typeof buf === 'string') {
-    buf = uint8ArrayFromString(buf, 'base16')
-  }
+    /**
+     * @type {PublicKey | false}
+     */
+    const pubKey = pubKeyArray ? await cryptoKeys.unmarshalPublicKey(pubKeyArray) : false
 
-  let { id, privKey, pubKey } = PeerIdProto.decode(buf)
+    /**
+     * @type {Uint8Array}
+     */
+    let pubDigest = new Uint8Array()
 
-  privKey = privKey ? await cryptoKeys.unmarshalPrivateKey(privKey) : false
-  pubKey = pubKey ? await cryptoKeys.unmarshalPublicKey(pubKey) : false
+    /**
+     * @type {Uint8Array}
+     */
+    let privDigest = new Uint8Array()
 
-  let pubDigest
-  let privDigest
+    if (privKey) {
+      // @ts-ignore
+      privDigest = await computeDigest(privKey.public)
+    }
 
-  if (privKey) {
-    privDigest = await computeDigest(privKey.public)
-  }
+    if (pubKey) {
+      pubDigest = await computeDigest(pubKey)
+    }
 
-  if (pubKey) {
-    pubDigest = await computeDigest(pubKey)
-  }
+    if (privKey) {
+      if (pubKey) {
+        if (!uint8ArrayEquals(privDigest, pubDigest)) {
+          throw new Error('Public and private key do not match')
+        }
+      }
+      // @ts-ignore
+      return new PeerId(privDigest, privKey, privKey.public)
+    }
 
-  if (privKey) {
+    // TODO: val id and pubDigest
     if (pubKey) {
-      if (!uint8ArrayEquals(privDigest, pubDigest)) {
-        throw new Error('Public and private key do not match')
-      }
+      return new PeerId(pubDigest, undefined, pubKey)
     }
-    return new PeerIdWithIs(privDigest, privKey, privKey.public)
-  }
 
-  // TODO: val id and pubDigest
+    if (id) {
+      return new PeerId(id)
+    }
 
-  if (pubKey) {
-    return new PeerIdWithIs(pubDigest, null, pubKey)
+    throw new Error('Protobuf did not contain any usable key material')
   }
 
-  if (id) {
-    return new PeerIdWithIs(id)
+  /**
+  * Checks if a value is an instance of PeerId.
+  * 
+  * @param {any} peerId - The value to check.
+  * @returns {boolean}
+  */
+  static isPeerId (peerId) {
+    return Boolean(typeof peerId === 'object' && peerId._id && peerId._idB58String)
```

Comment:
Let's use the Symbol here like https://github.com/multiformats/js-multiaddr/blob/master/src/index.js#L551